runtime/lua/vim/snippet.lua

```diff
@@ -101,6 +101,20 @@ local function get_extmark_range(bufnr, extmark_id)
   return { mark[1], mark[2], mark[3].end_row, mark[3].end_col }
 end
 
+local function set_keymaps()
+  vim.keymap.set({ 'i', 's' }, '<Tab>', function ()
+    return vim.snippet.active({ direction = 1 }) and '<cmd>lua vim.snippet.jump(1)<cr>' or '<Tab>'
+  end, { expr = true, silent = true })
+  vim.keymap.set({ 'i', 's' }, '<S-Tab>', function ()
+    return vim.snippet.active({ direction = -1 }) and '<cmd>lua vim.snippet.jump(-1)<cr>' or '<S-Tab>'
+  end, { expr = true, silent = true })
+end
+
+local function remove_keymaps()
```

Comment:
Not sure of what will happen if the user has a different keymap and we delete these ones here. Also not sure if we wanted to only set these keymaps if the user enabled completion...

Reply:
Good point; we'll have to save and restore them. (That's a bit more work than I envisaged but should be doable via `:h nvim_buf_get_keymap()` -- we only need one or two, after all)

Reply:
(Which reminds me: you'll want to set these mappings for the buffer only.)